Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -745,21 +389,23 @@ public extension ChatChannelController {
             self.messages.last { !$0.isLocalOnly }?.id
         }
 
-        guard let messageId = messageId ?? lastFetchedMessageId ?? lastLocalMessageId() else {
+        guard let messageId = messageId ?? lastOldestMessageId ?? lastLocalMessageId() else {
             log.error(ClientError.ChannelEmptyMessages().localizedDescription)
             callback { completion?(ClientError.ChannelEmptyMessages()) }
             return
         }
 
-        guard !hasLoadedAllPreviousMessages else {
-            completion?(nil)
+        guard !hasLoadedAllPreviousMessages && !isLoadingPreviousMessages else {
```

Comment:
This is on purpose. The completion call can be misleading here 🤔 A customer might think the page was actually loaded, which is not true. We skipped the fetch in this case.

Reply:
Worth adding a comment here to note this

Reply:
A method that has a completion block should always call it, otherwise unexpected results from the integration side can happen. IMO it is worth adding a descriptive error there.

Reply:
I'm not sure about this one. This method is called multiple times since the scroll position will be in the threshold multiple times. If we call the completion block **nil**, it might give the impression that the load was successful, although nothing really happened. If a developer wants to act when the load is successful, we won't have a way to distinguish between a real successful load or a skipped one. 

On the other hand, if we add an error, it could also be misleading since a customer might add an alert whenever loading more messages have an error. And since our API just returns a generic `Error` and not an enum, there's no easy way to act on a real error or just this "skipped" error. 

TL;DR: I prefer to add a comment and/or maybe a debug log here.

We could also say if this is actually a responsibility of the controller, or if it should be done in the UI. Previously it was done in the UI, but I thought that moving this to the Controller would make it less "boilerplaty"